pkg/machine/applehv/machine.go

```diff
@@ -186,9 +187,40 @@ func (m *MacMachine) writeIgnitionConfigFile(opts machine.InitOptions, key strin
 
 func (m *MacMachine) Init(opts machine.InitOptions) (bool, error) {
 	var (
-		key          string
-		virtiofsMnts []machine.VirtIoFs
+		key                string
+		virtiofsMnts       []machine.VirtIoFs
+		cleanCallbackFuncs []func() error
+		err                error
 	)
+
+	// cleanup half-baked files if init fails at any point, or if init is sigterm'd
+	cleanCallback := func() {
+		for _, clean := range cleanCallbackFuncs {
+			if err := clean(); err != nil {
+				logrus.Error(err)
+			}
+		}
+	}
+	defer func() {
+		if err != nil {
+			cleanCallback()
+		}
+	}()
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
```

Comment:
is the signal logic needed here or does it make things more complex say versus a bool?

Reply:
This is in case someone ctrl-c's during init, not necessarily for failure.